src/functions.php

```diff
@@ -285,25 +333,14 @@ function is_last_pipeline_operator_write(array $pipeline): bool
  * @internal
  * @see https://mongodb.com/docs/manual/reference/command/mapReduce/#output-inline
  * @param string|array|object $out Output specification
- * @throws InvalidArgumentException
```

Comment:
I originally removed this because `is_mapreduce_output_inline()` starts with:

```php
if (! is_array($out) && ! is_object($out)) {
    return false;
}
```

`document_to_array()` only throws if the final `$document` value is not an array of object, so I assumed an exception wouldn't be possible.

Thinking about this some more, I suppose we could still throw if a Serializable instance were to return something other than an array or object from `bsonSerialize()`. Granted, that would break in other ways, but that's beyond the scope of ``is_mapreduce_output_inline()`.

In that case, it's still possible for `is_pipeline()` to throw for the same reason (i.e. a Serializable object in the pipeline array returns a bad value).

Note: I think this edge case only applies to PHP < 7.x, since `bsonSerialize()` has a return type hint for PHP 8.0: https://github.com/mongodb/mongo-php-driver/blob/1.15.3/src/BSON/Serializable.stub.php#L12-L18